Fix flaky test `test_get_dashboard`

I have run into an intermittent test failures on Travis on my different PRs. Because `list` does not maintain its order, this test fails intermittently.

**Link to test**
https://github.com/mitodl/micromasters/blob/master/dashboard/views_test.py#L77-L85

**Staketrace** 
```
self = <dashboard.views_test.DashboardTest testMethod=test_get_dashboard>
    def test_get_dashboard(self):
        """Test for GET"""
        with patch('backends.utils.refresh_user_token', autospec=True):
            result = self.client.get(self.url)
        assert 'programs' in result.data
        assert 'is_edx_data_fresh' in result.data
        assert result.data['is_edx_data_fresh'] is True
        assert len(result.data['programs']) == 2
>       assert [self.program_1.id, self.program_2.id] == [res_item['id'] for res_item in result.data['programs']]
E       AssertionError: assert [161, 162] == [162, 161]
E         At index 0 diff: 161 != 162
E         Use -v to get the full diff
dashboard/views_test.py:85: AssertionError
```

FYI @pdpinch

--- a/dashboard/views_test.py
+++ b/dashboard/views_test.py
@@ -82,7 +82,8 @@ class DashboardTest(MockedESTestCase, APITestCase):
         assert 'is_edx_data_fresh' in result.data
         assert result.data['is_edx_data_fresh'] is True
         assert len(result.data['programs']) == 2
-        assert [self.program_1.id, self.program_2.id] == [res_item['id'] for res_item in result.data['programs']]
+        assert set([self.program_1.id, self.program_2.id]) == set(
+            [res_item['id'] for res_item in result.data['programs']])
 
     @ddt.data(Instructor, Staff)
     @patch('dashboard.api.CachedEdxDataApi.update_cache_if_expired')

